Nodejs/Product/ProjectWizard/ImportWizard.cs

```diff
@@ -15,74 +14,75 @@ namespace Microsoft.NodejsTools.ProjectWizard
-                Guid packageGuid = new Guid(Guids.NodejsPackageString);
-                IVsPackage package;
-                ((IVsShell)Package.GetGlobalService(typeof(SVsShell))).LoadPackage(
-                    ref packageGuid,
-                    out package
-                );
+                var hr = EnsurePackageLoaded(serviceProvider);
+                if (ErrorHandler.Failed(hr))
+                {
+                    MessageBox.Show(string.Format(ProjectWizardResources.ImportWizardCouldNotStartFailedToLoadPackage, hr), "Visual Studio");
+                    throw new WizardBackoutException();
+                }
+                var uiShell = (IVsUIShell)serviceProvider.GetService(typeof(SVsUIShell));
 
-                System.Threading.Tasks.Task.Factory.StartNew(() =>
+                var projName = replacementsDictionary["$projectname$"];
+                replacementsDictionary.TryGetValue("$specifiedsolutionname$", out var solnName);
+                string directory;
+                if (string.IsNullOrWhiteSpace(solnName))
+                {
+                    // Create directory is unchecked, destinationdirectory is the
+                    // directory name the user entered plus the project name, we want
+                    // to remove the solution directory.
+                    directory = Path.GetDirectoryName(replacementsDictionary["$destinationdirectory$"]);
+                }
+                else
                 {
-                    string projName = replacementsDictionary["$projectname$"];
-                    string solnName = replacementsDictionary["$specifiedsolutionname$"];
-                    string directory;
-                    if (string.IsNullOrWhiteSpace(solnName))
-                    {
-                        // Create directory is unchecked, destinationdirectory is the
-                        // directory name the user entered plus the project name, we want
-                        // to remove the solution directory.
-                        directory = Path.GetDirectoryName(replacementsDictionary["$destinationdirectory$"]);
-                    }
-                    else
-                    {
-                        // Create directory is checked, the destinationdirectory is the
-                        // directory the user entered plus the project name plus the
-                        // solution name.
-                        directory = Path.GetDirectoryName(replacementsDictionary["$destinationdirectory$"]);
-                    }
+                    // Create directory is checked, the destinationdirectory is the
+                    // directory the user entered plus the project name plus the
+                    // solution name - we want to remove both extra folders
+                    directory = Path.GetDirectoryName(Path.GetDirectoryName(replacementsDictionary["$destinationdirectory$"]));
```

Comment:
Presumably, there's some reason to believe `$destinationdirectory$` has a grandparent?